src/System.Private.CoreLib/src/System/Runtime/InteropServices/PInvokeMarshal.Unix.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// This PInvokeMarshal class should provide full public Marshal 
+    /// implementation for all things related to P/Invoke marshalling
+    /// </summary>
+    public partial class PInvokeMarshal
+    {
+        public static void SaveLastWin32Error()
+        {
+            s_lastWin32Error = Interop.Sys.GetLastErrNo();
+        }
+
+        internal static void ClearLastWin32Error()
+        {
+            Interop.Sys.SetLastErrNo(s_lastWin32Error);
```

Comment:
This should be setting it to 0, like what it does in CoreCLR:

https://github.com/dotnet/coreclr/blob/32f0f9721afb584b4a14d69135bea7ddc129f755/src/vm/stubhelpers.cpp#L1171